internal/cache/async.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+
+	"github.com/palantir/k8s-spark-scheduler/internal/cache/store"
+	"github.com/palantir/witchcraft-go-logging/wlog/svclog/svc1log"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	rest "k8s.io/client-go/rest"
+)
+
+type object interface {
+	metav1.Object
+	runtime.Object
+}
+
+type asyncClient struct {
+	client             rest.Interface
+	resourceName       string
+	emptyObjectCreator func() object
```

Comment:
seems weird

Reply:
There is no interface both metav1.Object and runtime.Object satisfy unfortunately

Reply:
now using an interface to abstract types out from the clients